src/libcollections/dlist.rs

```diff
@@ -1002,6 +1138,39 @@ mod tests {
     }
 
     #[test]
+    fn test_split_off() {
+        // singleton
+        {
+            let mut m = DList::new();
+            m.push_back(1i);
+
+            let n = m.split_off(1);
+            assert_eq!(m.len(), 1);
+            assert_eq!(n.len(), 0);
+            assert_eq!(m.front(), Some(1));
+
+            let p = m.split_off(0);
+            assert_eq!(m.len(), 0);
+            assert_eq!(p.len(), 1);
+            assert_eq!(p.back(), Some(1));
+            assert_eq!(p.front(), Some(1));
+        }
+
+        // not singleton
+        let mut m = vec![1i,2,3,4,5];
+        let mut n = m.split_off(2);
+        assert_eq!(m.len(), 2);
+        assert_eq!(n.len(), 3);
+        for elt in range(1i, 3) {
+            assert_eq!(m.pop_front(), elt);
+        }
+        for elt in range(3i, 6) {
+            assert_eq!(n.pop_front(), elt);
+        }
+
```

Comment:
Could you have this cover both "should walk from back" and "should walk from front"?